lib/c.c

```diff
@@ -370,12 +370,89 @@ int fputc(int c, FILE *stream)
  * FIXME: adopt lite_malloc from musl-libc
  *   https://git.musl-libc.org/cgit/musl/tree/src/malloc/lite_malloc.c
  */
-void *malloc(int size)
+
+typedef struct block_meta {
+    int size;
+    struct block_meta *next;
+    int free;
+} block_meta_t;
+
+block_meta_t *global_base;
+block_meta_t *global_last;
```

Comment:
Add pseudo namespace to avoid potential symbol conflicts. That is, rename the variables to something like `__malloc_global_base`.